autogpts/autogpt/autogpt/config/config.py

```diff
@@ -297,36 +301,54 @@ def build_config_from_env(cls, project_root: Path = PROJECT_ROOT) -> Config:
         return config
 
 
-def assert_config_has_openai_api_key(config: Config) -> None:
+async def assert_config_has_openai_api_key(config: Config) -> None:
     """Check if the OpenAI API key is set in config.py or as an environment variable."""
-    if not config.openai_credentials:
-        print(
-            Fore.RED
-            + "Please set your OpenAI API key in .env or as an environment variable."
-            + Fore.RESET
+    key_pattern = r"^sk-\w{48}"
+    openai_api_key = (
+        config.openai_credentials.api_key.get_secret_value()
+        if config.openai_credentials and config.openai_credentials.api_key
```

Comment:
Redundant check; `api_key` is a mandatory attribute on the `OpenAICredentials` model